### PR TITLE
Add features dynamic actions

### PR DESCRIPTION
**Pull request by younesStrittmatter**
Added dynamic versioning:
- add option to choose
- update pyproject toml to include it
- update readme
Added action to publish
- add option to choose
- update readme
- add files if chosen (and delete them again if not in post_gen)

### CODE REVIEW

File: {{ cookiecutter.__project_slug }}/README.md [hunk 1]
```diff
@@ -29,28 +29,25 @@ pip install -e ".[dev]"
...
-{% if cookiecutter.autora_contribution_type == "theorist" %}
-### **Theorist**
+## Contribution Guide
+{% if cookiecutter.autora_contribution_type == "theorist" -%}
+### Theorist
 An sklearn regressor that returns an interpretable model relating experiment conditions $X$ to 
 observations $Y$.<br>
 *Example: The [Bayesian Machine Scientist](https://github.com/AutoResearch/autora-theorist-bms) (Guimerà et al., 2020, 
 in Science Advances) returns an equation governing the relationship between $X$ and $Y$.* <br>
 {% endif %}
```
Comment (benwandrew) [inline {{ cookiecutter.__project_slug }}/README.md:39]:
```suggestion
{% endif -%}
```

Comment (benwandrew) [reply {{ cookiecutter.__project_slug }}/README.md:39]: not sure i understand the syntax, but this should conform with the rest, no?

File: {{ cookiecutter.__project_slug }}/README.md [hunk 2]
```diff
@@ -124,20 +118,21 @@ the package is still in progress.
...
-There are at least two options for publishing the package. For beginners, we recommend Option 1 (via Github Actions) 
-as it is easier to follow.
+{% if cookiecutter.use_github_actions == "yes" -%}
+#### Step 5.2 Publish via GitHub Actions
 
-#### Step 5.2 (Option 1): Publish via GitHub Actions
-
-To automate the publishing process for your package, you can use a GitHub action instead of Twine:
+To automate the publishing process for your package, you can use a GitHub action:
+- Add a github secret in your repository named PYPI_API_TOKEN, that contains a pyPI token of your account
```
Comment (benwandrew) [inline {{ cookiecutter.__project_slug }}/README.md:135]:
```suggestion
- Add a github secret in your repository named PYPI_API_TOKEN, that contains a PyPI token of your account
```

**Review by benwandrew — CHANGES_REQUESTED** [review]
nice! one question and one nitpick.

**Comment by younesStrittmatter** [comment]
{% ... %} automatically inserts a empty line. This leads to a lot of empty lines that we don't want. To prevent this from happening, we can use {% ... -%} instead.

**Review by benwandrew — APPROVED** [review]
excellent!

**Pull request merged**